drafts/Stormless.md

```diff
@@ -0,0 +1,198 @@
+"email": "santa@mad.russian.xmas.com"}`, the event passed to the Lambda is:
+
+```
+...
+"body": {
+    "first_name": "Santa", 
+    "email": "santa@mad.russian.xmas.com"
+}
+```
+Knowing the input data structure is important to map it to the action input parameters. It's intuitive: `input` represents `event` parameter of [AWS Lambda programming model](http://docs.aws.amazon.com/lambda/latest/dg/python-programming-model-handler-types.html) (BTW should we call it `event`? Vote with a PR!). [Jinja](http://jinja.pocoo.org/) is used to map the inputs; our JavaScript friends who're less familiar
+with this common Python tool find it intuitive enough in simple cases; and Stack-overflow is full of magic Jinga tricks.
+
+Here we obviously map the two parameters from input body to desired action input parameters. 
+
+To keep the config separately, I created a file `env.yml` and put my config parameters there:
+
+```
+# ./env.yml
+# WARNING: Don't commit to Github :)
+slack:
+  admin_token: "xoxs-111111111111-111111111111-111111111111-1111111111"
+  organization: "my-team"
+
+```
+Than I used it in `serverless.yml` like this: `admin: ${file(env.yml):slack}`. It's self-explanatory. 
+
+Optionally, we can also form a function output from action results. Let's keep it simple for now, I'll share tricks later.
+
+Ok, that's it! The function is ready to fly to AWS. But I take it sloooow. First, I'll package it locally. 
+
+```
+sls -package
+```
+The first time takes a long while as this is the time when the plugin sets up itself. It will pull the Docker images from the Hub (oh, did I say you got to be connected? Or we assume internet connection a basic commodity like briefing air and electric power?). It will install StackStorm runners - the code that knows to run StackStorm exchange packs. It will pull the `slack` pack from Exchange. It will install `slack` pack python dependencies. It takes a long time the first time. Good news: it's only the first time. 
+
+Now let's run this locally.
+
+```
+sls stackstorm docker run --function InviteSlack --verbose --data '{"body": {"first_name": "Santa", 
+"email": "santa@mad.russian.xmas.com"}}'
+```
+
+Local runs in the container. It takes a bit longer, but ensures that the execution environment matches AWS lambda very closely, so better safe than sorry. 
+
+If I want to 
+
+Now we are really ready ready. Deploy the function to AWS, and run it "serverless".
+
+```
+sls deploy
+```
+
+It will take some while - now it's serverless (and honestly, our bundle is a bit bloated, patience! plugin developers are currently busy solving other problems, we will optimize it as soon as we can)
+
+
+> PRO TIP: if something goes wrong at this point, most likely your AWS setup is fucked up. Go to Getting Ready, step 2. Google, Stack-overflow.
```

Comment:
I am assuming you didn't mean to keep this in here :) I am pretty `-1` to including profanity in blog content.

Reply:
:joy:

Reply:
That leaked already: committed to history and pushed on Github. 
No, I won't post it :)